src/pages/resultsView/SurvivalStoreHelper.ts

```diff
@@ -50,3 +51,46 @@ export function getPatientSurvivals(
         return [];
     }
 }
+
+export function getPatientSurvivalUniqueData(
+    survivalClinicalDataGroupByUniquePatientKey: any,
+    targetUniquePatientKeys: string[],
+    statusAttributeId: string,
+    monthsAttributeId: string
+): string[] {
+    if (targetUniquePatientKeys) {
+        return targetUniquePatientKeys.reduce(
+            (patientSurvivalData: string[], uniquePatientKey: string) => {
+                const clinicalData: ClinicalData[] =
+                    survivalClinicalDataGroupByUniquePatientKey[
+                        uniquePatientKey
+                    ];
+                if (clinicalData) {
+                    const statusClinicalData:
+                        | ClinicalData
+                        | undefined = clinicalData.find(
+                        c => c.clinicalAttributeId === statusAttributeId
+                    );
+                    const monthsClinicalData:
+                        | ClinicalData
+                        | undefined = clinicalData.find(
+                        c => c.clinicalAttributeId === monthsAttributeId
+                    );
+                    if (
+                        statusClinicalData &&
+                        monthsClinicalData &&
+                        statusClinicalData.value != 'NA' &&
+                        monthsClinicalData.value != 'NA' &&
+                        !Number.isNaN(Number(monthsClinicalData.value))
+                    ) {
+                        patientSurvivalData.push(statusClinicalData.value);
+                    }
+                }
+                return _.uniq(patientSurvivalData);
```

Comment:
i think you're calling uniq inside the reduce loop.  why not call it once on final result of reduce?